src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,87 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { patternLetterAndNumbers, patternUrl } from '../../constants/constants';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, addCount] = useState(1);
+  const [state, setState] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handlerStateFilm = (key: string, value: string) => {
+    setState({ ...state, [key]: value });
+  };
+
+  const error = state.title.trim()
+  && patternLetterAndNumbers.test(state.imdbId.trim())
```

Comment:
and here delete patternLetterAndNumbers as well, imdbId
```suggestion
  && state.imdbId.trim()
```